diff --git a/dist/cjs/node/server/ServerSubscription.js b/dist/cjs/node/server/ServerSubscription.js
index 00511a1045a387e1c7a57d42a323b9092795df47..a5d7169ea0ae65c39c0df8420b46ed3d76ec9e12 100644
--- a/dist/cjs/node/server/ServerSubscription.js
+++ b/dist/cjs/node/server/ServerSubscription.js
@@ -667,6 +667,14 @@ class ServerSubscription {
     }
   }
   async #sendUpdateMessage(lifetime, attributeFilter, eventsMinNumber, onlyWithData, session) {
+    if (session === void 0) {
+      const ownSession = this.#context.session;
+      if (ownSession !== void 0 && !ownSession.isClosing && !ownSession.isClosed) {
+        session = ownSession;
+      } else {
+        return false;
+      }
+    }
     const exchange = this.#context.initiateExchange(session ?? this.#peerAddress, import_types.INTERACTION_PROTOCOL_ID);
     if (exchange === void 0) {
       return false;
diff --git a/dist/esm/node/server/ServerSubscription.js b/dist/esm/node/server/ServerSubscription.js
index 28bfefa2b9bafeb3ce12d058d7833b3d29fcff87..cfa5c2736493ceb89ec7e513f91161bcc43c4d33 100644
--- a/dist/esm/node/server/ServerSubscription.js
+++ b/dist/esm/node/server/ServerSubscription.js
@@ -671,6 +671,14 @@ class ServerSubscription {
     }
   }
   async #sendUpdateMessage(lifetime, attributeFilter, eventsMinNumber, onlyWithData, session) {
+    if (session === void 0) {
+      const ownSession = this.#context.session;
+      if (ownSession !== void 0 && !ownSession.isClosing && !ownSession.isClosed) {
+        session = ownSession;
+      } else {
+        return false;
+      }
+    }
     const exchange = this.#context.initiateExchange(session ?? this.#peerAddress, INTERACTION_PROTOCOL_ID);
     if (exchange === void 0) {
       return false;
