diff --git a/dist/cjs/node/server/ServerSubscription.js b/dist/cjs/node/server/ServerSubscription.js
index ccbdfe942cd237b262a06ce1506277246ee8399a..7ce1bb5d8600b02982a07f725a42113aea8a9758 100644
--- a/dist/cjs/node/server/ServerSubscription.js
+++ b/dist/cjs/node/server/ServerSubscription.js
@@ -660,6 +660,14 @@ class ServerSubscription {
     }
   }
   async #sendUpdateMessage(lifetime, attributeFilter, eventsMinNumber, onlyWithData, session) {
+    if (session === void 0) {
+      const ownSession = this.#context.session;
+      if (ownSession !== void 0 && !ownSession.isClosing && !ownSession.isClosed) {
+        session = ownSession;
+      } else {
+        return false;
+      }
+    }
     const exchange = this.#context.initiateExchange(session ?? this.#peerAddress, import_types.INTERACTION_PROTOCOL_ID);
     if (exchange === void 0) {
       return false;
diff --git a/dist/esm/node/server/ServerSubscription.js b/dist/esm/node/server/ServerSubscription.js
index d74af17cbc673249e63e5452327bf37e73061cf4..4fb9ba33e9d31cff9bd41371bd4b71fe9040f484 100644
--- a/dist/esm/node/server/ServerSubscription.js
+++ b/dist/esm/node/server/ServerSubscription.js
@@ -664,6 +664,14 @@ class ServerSubscription {
     }
   }
   async #sendUpdateMessage(lifetime, attributeFilter, eventsMinNumber, onlyWithData, session) {
+    if (session === void 0) {
+      const ownSession = this.#context.session;
+      if (ownSession !== void 0 && !ownSession.isClosing && !ownSession.isClosed) {
+        session = ownSession;
+      } else {
+        return false;
+      }
+    }
     const exchange = this.#context.initiateExchange(session ?? this.#peerAddress, INTERACTION_PROTOCOL_ID);
     if (exchange === void 0) {
       return false;
